Test/dafny0/TypeConstraints.dfy

```diff
@@ -415,20 +415,46 @@ module LetPatterns {
 }
 
 module Arrays_and_SubsetTypes {
+  method M()
+  {
+    var a: array<nat>;
+    var b: array<int>;
+    if * {
+      a := new nat[100];
+      b := new nat[100]; // ERROR
+    } else if * {
+      a := new int[100]; // ERROR
+      b := new int[100];
+    } else if * {
+      a := b;  // ERROR
+    } else if * {
+      b := a;  // ERROR
+    } else if * {
+      var n := new nat[100];  // array<nat>
+      if * {
+        a := n;
+      } else {
+        b := n;
```

Comment:
This should also be an error

Reply:
It is a verification error. Moved it to a different file to check it.